### ISSUE TITLE
Bad build → Cannot read property 'props' of undefined

### ISSUE THREAD

**Issue by pushred** [gilbarbara/react-joyride#435]
>Expected behavior

Using react-joyride with create-react-app 1.5.2 (latest) just works

>Actual behavior

https://codesandbox.io/s/9z8456yyxy

Trying to render the component fails with an exception.

>Steps to reproduce the problem

```
npx create-react-app react-joyride.test
cd react-joyride.test
npm install react-joyride
```

Change default `App.js` component to:

```jsx
class App extends Component {
  constructor() {
    super();
    this.state = { joyrideActive: false };
  }

  componentDidMount() {
    this.setState({ joyrideActive: true })
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
        <Joyride
          steps={[{
            target: '.App-intro',
            content: 'Get started!',
            placement: 'bottom',          
          }]}
          run={this.state.joyrideActive}
        />
      </div>
    );
  }
}
```

>React version

16.4.2

>React-Joyride version

1.11.4

>Browser name and version

Chrome 69.0.3497.100

>Error stack (if available)

```
node_modules/react-joyride/es/index.js:797
  794 |     type: 'joyride:checkStepValidity',
  795 |     msg: ['Provided a step without the required ' + requiredField + ' property.', 'Step:', step],
  796 |     warn: true,
> 797 |     debug: _this4.props.debug
  798 |   });
  799 | }
  800 | 
```

Same issue when trying to import from `lib`

```
node_modules/react-joyride/lib/index.js:820
  817 |     type: 'joyride:checkStepValidity',
  818 |     msg: ['Provided a step without the required ' + requiredField + ' property.', 'Step:', step],
  819 |     warn: true,
> 820 |     debug: _this4.props.debug
  821 |   });
  822 | }
  823 | 
```

> Steps to a working build

- Clone this repo
- `npm run build`
- npm link
- npm link in CRA project

Differences in build output can be seen here: https://gist.github.com/pushred/7d8a4df145a2cc137b47e21bf4436c4b

Working version is the 2nd commit.

I'm on node 6.14.4, the differences are maybe related to babel-preset-env or the published build is just incorrect for some other reason.

**Comment by richardaum** [comment]
It seems version 1.11.4 is broken. [This](https://codesandbox.io/s/2zpjporp4p) tutorial is using a higher version, like 2.0.0.

**Comment by dherault** [comment]
I'm getting the same error.

**Comment by JimiHFord** [comment]
As @richardaum pointed out, it is an issue with `1.11.4` (not sure if any earlier versions are affected or not).

The README suggests installing like so `npm i react-joyride@next`. This pulls in `^2.0.0-x` and fixed the issue for me.

It would be nice if the package was updated in npm so that `2.0.0-x` is the latest. As of right now, npm shows `1.11.4` as the current version. 

The [GitHub project's Releases page](https://github.com/gilbarbara/react-joyride/releases) shows `1.11.2` as the latest version which should be updated as well.

**Comment by JimiHFord** [comment]
Whoops - looks like gilbarbara has [intentionally not released 2.0 yet](https://github.com/gilbarbara/react-joyride/issues/291#issuecomment-405089465). Seems like there is still some confusion in the community around what the current version is though.

**Comment by gilbarbara** [comment]
2.0 will be released this week
